src/OrbitQt/TrackTypeItemModel.h

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_QT_TRACK_TYPE_ITEM_MODEL_H_
+#define ORBIT_QT_TRACK_TYPE_ITEM_MODEL_H_
+
+#include <QAbstractTableModel>
+#include <QIcon>
+#include <QModelIndex>
+#include <QObject>
+#include <QString>
+#include <QVariant>
+#include <QtCore>
+#include <array>
+#include <map>
+
+#include "Track.h"
+#include "TrackManager.h"
+
+namespace orbit_qt {
+
+class TrackTypeItemModel : public QAbstractTableModel {
+  Q_OBJECT
+ public:
+  enum class Column { kVisibility, kName, kEnd };
+
+  explicit TrackTypeItemModel(QObject* parent = nullptr);
+
+  [[nodiscard]] int columnCount(const QModelIndex& parent = {}) const override;
+  [[nodiscard]] QVariant data(const QModelIndex& idx, int role = Qt::DisplayRole) const override;
+  [[nodiscard]] bool setData(const QModelIndex& idx, const QVariant& value,
+                             int role = Qt::EditRole) override;
+  [[nodiscard]] QVariant headerData(int section, Qt::Orientation orientation,
+                                    int role = Qt::DisplayRole) const override;
+  [[nodiscard]] int rowCount(const QModelIndex& parent = {}) const override;
+  [[nodiscard]] Qt::ItemFlags flags(const QModelIndex& index) const override;
+
+  void SetTrackManager(TrackManager* track_manager);
+
+ private:
+  TrackManager* track_manager_ = nullptr;
+  const std::vector<Track::Type> known_track_types_;
```

Comment:
Regarding the previous discussion on this, `const` member variables are better avoided. For example, they make a type non-moveable. If this is supposed to be fixed it can be `static`, otherwise it shouldn't be `const`.

Reply:
As discussed, I don't want to make them `static` because I see them conceptually related to the instance, even though right now you can't change them from the outside. Valid point about movability, I've removed the `const` now.